code/go/0chain.net/smartcontract/dbs/event/sharder.go

```diff
@@ -180,6 +181,43 @@ func (sh *Sharder) exists(edb *EventDb) (bool, error) {
 	return true, nil
 }
 
+type SharderQuery struct {
+	gorm.Model
+	SharderID         null.String
+	N2NHost           null.String
+	Host              null.String
+	Port              null.Int
+	Path              null.String
+	PublicKey         null.String
+	ShortName         null.String
+	BuildTag          null.String
+	TotalStaked       null.Int
+	Delete            null.Bool
+	DelegateWallet    null.String
+	ServiceCharge     null.Float
+	NumberOfDelegates null.Int
+	MinStake          null.Int
+	MaxStake          null.Int
+	LastHealthCheck   null.Int
+	Rewards           null.Int
+	Fees              null.Int
+	Active            null.Bool
+	Longitude         null.Int
+	Latitude          null.Int
```

Comment:
have never used null
what benefit does it bring?

Reply:
`Int is an nullable int64. It does not consider zero values to be null. It will decode to null, not zero, if null.` 
Same for Bool too, by default bool is assigned false

Reply:
@kushthedude still unclear
can you state with an example.

Reply:
@ssardana08 while filtering we want values or we don't. 
So for example if we need Longitude = 0 and we do not use null.Int the form by default will not include `where Longitude=0` in query. So we use nullable value so the gorm is sure about whether the value is assigned intentionally or is just some default value. So if you now set `longitude=0` gorm will include the value in the query.

Reply:
great!